lms/djangoapps/discussion/rest_api/permissions.py

```diff
@@ -148,3 +153,29 @@ def has_permission(self, request, view):
             CourseEnrollment.is_enrolled(request.user, course_key) or
             has_discussion_privileges(request.user, course_key)
         )
+
+
+class IsStaffOrCourseTeamOrAuthor(permissions.BasePermission):
+    """
+    Permission that checks to see if the user is allowed to request
+    content authored by an user in a specific course.
+    """
+
+    def has_permission(self, request, view):
+        """
+        Returns True if the user is staff or an enrolled user requesting
+        their own content.
+        """
+        course_key = CourseKey.from_string(view.kwargs.get("course_id"))
+        return (
+            GlobalStaff().has_user(request.user) or
+            CourseStaffRole(course_key).has_user(request.user) or
+            CourseInstructorRole(course_key).has_user(request.user) or
+            (
+                (
+                    CourseEnrollment.is_enrolled(request.user, course_key) or
+                    has_discussion_privileges(request.user, course_key)
+                ) and
+                request.user.username == view.kwargs.get("username")
+            )
+        )
```

Comment:
I'm not sure this is needed. This should be accessible by anyone enrolled in the course.

Reply:
The flagged filter should only be usable by those with preveleges though.